README.md

```diff
@@ -1 +1,100 @@
 # TDR Statuses
+
+This lambda takes file information as an input and produces a list of file and consignment statuses.
+
+## File Statuses
+
+Given this json as input:
+```json
+
+{
+  "results": [
+    {
+      "fileId": "20d80488-d247-47cf-8687-be26de2558b5",
+      "originalPath": "smallfile/subfolder/subfolder-nested/subfolder-nested-1.txt",
+      "fileSize": "2",
+      "clientChecksum": "87428fc522803d31065e7bce3cf03fe475096631e5e07bbd7a0fde60c4cf25c7",
+      "consignmentType": "standard",
+      "fileCheckResults": {
+        "antivirus": [
+          {
+            "result": ""
+          }
+        ],
+        "checksum": [
+          {
+            "sha256Checksum": "87428fc522803d31065e7bce3cf03fe475096631e5e07bbd7a0fde60c4cf25c7"
+          }
+        ],
+        "fileFormat": [
+          {
+            "matches": [
+              {
+                "puid": "fmt/866"
+              }
+            ]
+          }
+        ]
+      }
+    }
+  ],
+  "redactedResults": {
+    "redactedFiles": [
+      {
+        "originalFileId": "4dbabaef-1fae-4d09-8faa-88e9dfb85b05",
+        "originalFilePath": "originalPath",
+        "redactedFileId": "7daa4ab6-ab7d-449a-88f5-a8ff1705b888",
+        "redactedFilePath": "redactedPath"
+      }
+    ],
+    "errors": [
+      {
+        "fileId": "97536958-2192-4485-af9a-d98a0290e692",
+        "cause": "FailureReason"
+      }
+    ]
+  }
+}
+```
+
+The lambda processes the following file statuses for each file.
+
+### Antivirus
+* `Success` if antivirus result is empty
+* `VirusDetected` if the result is not empty
+
+### FFID
+* If the consignment type is judgment and the puid is not in the AllowedPuids table then `NonJudgmentFormat`.
+* If the consignment type is standard and the puid is in the DisallowedPuids table then get the status from the `Reason`  column.
+* If the file size is zero then `ZeroByteFile`.
+* Otherwise `Success`.
```

Comment:
So I've changed this now so the FileStatus will be set with the "Reason" from the "DisallowedPuids" table whether it is active or not.
Then for the "ConsignmentStatus", it will check the file statuses for that consignment and if those file statuses are either `Success` or one of  the inactive reasons, it returns `Completed` otherwise it's completed with issues.

I've not mentioned `PasswordProtected` and `Zip` in the README explicitly because these are driven from the database so they're covered in the "status from the `Reason`  column" point.

Reply:
OK, that all makes sense now. There is just one outstanding comment about the RDS certificate, which may or may not be idiocy.